core/src/main/java/hudson/model/Run.java

```diff
@@ -1435,39 +1467,60 @@ public String toString() {
         return rawF;
     }
 
+    /**
+     * Gets Log file.
+     * @deprecated Not all {@link jenkins.model.logging.LogStorage} implementations
+     * are able to produce the log file efficiently.
+     * It is recommended to use {@link #getLogStorage()} to get browser with better API options.
+     * @return Log file.
+     *         It is never {@code null}, but the file may not exist
+     */
+    @Deprecated
+    @Nonnull
+    public File getLogFile() {
+        try {
+            return getLogStorage().getLogFile();
+        } catch (IOException|InterruptedException e) {
+            LOGGER.log(Level.WARNING, "Failed to locate log file for " + this, e);
+            return new File(getRootDir(), "log");
+        }
+    }
+
+    /**
+     * Deletes the log in the storage.
+     * @return {@code true} if the log was deleted.
+     *         {@code false} if Log deletion is not supported.
+     * @throws IOException Failed to delete the log.
+     * @throws InterruptedException Operation was interrupted
+     * @since TODO
+     */
+    public boolean deleteLog() throws IOException, InterruptedException {
+        return getLogStorage().deleteLog();
+    }
+
     /**
      * Returns an input stream that reads from the log file.
      * It will use a gzip-compressed log file (log.gz) if that exists.
      *
-     * @throws IOException 
+     * @throws IOException Operation error
      * @return An input stream from the log file. 
      *   If the log file does not exist, the error message will be returned to the output.
      * @since 1.349
      */
     public @Nonnull InputStream getLogInputStream() throws IOException {
-    	File logFile = getLogFile();
```

Comment:
For more current ideas about these methods, see overloads in `WorkflowRun` edited during https://github.com/jenkinsci/workflow-job-plugin/pull/114 etc.